src/libraries/System.Configuration.ConfigurationManager/src/System/Configuration/SettingsPropertyValue.cs

```diff
@@ -95,11 +95,11 @@ private object Deserialize()
                     {
                         value = GetObjectFromString(Property.PropertyType, Property.SerializeAs, (string)SerializedValue);
                     }
-                    else
+                    else if (SerializedValue is byte[] serializedBytes)
                     {
                         if (SettingsProperty.EnableUnsafeBinaryFormatterInPropertyValueSerialization)
                         {
-                            using (MemoryStream ms = new MemoryStream((byte[])SerializedValue))
+                            using (MemoryStream ms = new MemoryStream(serializedBytes))
```

Comment:
Not related to your change, but is anything in the catch block below still relevant?

Reply:
That's a good point.  This implementation only ever runs on .NETCore, and noone defines `RaisePropertyDeserializationWebErrorEvent` any more.  We can just delete it that - but this still needs the catch-all exceptions path.  I'll make a follow up PR and we can decide when to take it.